### PR TITLE
Replace deprecated `actions-rs/*` GH Actions

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Other

**Required reviews:** 
- 1

**What this does:**
- See this issue for details: https://github.com/nucypher/ferveo/issues/153
- Related: https://github.com/nucypher/nucypher-core/pull/90

### CODE REVIEW

**Review by theref — APPROVED** [review]

File: umbral-pre/src/bindings_python.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ fn hash(data: impl AsRef<[u8]>) -> i64 {
     // we just need something fast that minimizes conflicts.
     let digest = Sha256::new().chain(data).finalize();
     let (chunk, _): (GenericArray<u8, U8>, _) = digest.split();
-    let arr: [u8; 8] = chunk.try_into().unwrap();
+    let arr: [u8; 8] = chunk.into();
```
Comment (manumonti) [inline umbral-pre/src/bindings_python.rs:62]: Just out of curiosity: is this change related to these GH Actions in some way?

**Review by manumonti — APPROVED** [review]
LGTM! 👌

File: umbral-pre/src/bindings_python.rs (hunk 1 above)
Comment (piotr-roslaniec) [reply umbral-pre/src/bindings_python.rs:62]: I think it's an accidental change. It's possible that `clippy` was updated between the previous CI run and now, and it resulted in a broken linter rule.
